docuware-connector/src/com/axonivy/connector/docuware/connector/DocuWareService.java

```diff
@@ -55,4 +95,139 @@ public Date stringToDate(String dateString) {
+		contentType = Boundary.addBoundary(contentType);
+
+		Response response = prepareRestClient(target, configuration)
+				.post(Entity.entity(multipart, contentType));
+
+		FileUtils.forceDelete(propertiesFile.getJavaFile());
+		Document document = null;
+		
+		if(Status.Family.SUCCESSFUL == response.getStatusInfo().getFamily()) {	
+			 document = response.readEntity(Document.class) ;
+		 }else{			
+			 DocuWareException exception = handleError(response);
+			 throw exception;
+		}
+		response.close();
+		return document;
+	}
+
+	public static DocuWareException handleError(Response response) {
+		String errXml = response.readEntity(String.class);
+		String httpStatus = String.valueOf(response.getStatus());
+		String msg = "DocuWare Service call failed";
+		DocuWareException exception = new DocuWareException(msg, httpStatus);
+		try{
+			InputStream isr = new ByteArrayInputStream(errXml.getBytes(StandardCharsets.UTF_8));
+			DocumentBuilderFactory dbf = DocumentBuilderFactory.newInstance();
+			DocumentBuilder db = dbf.newDocumentBuilder();
+			org.w3c.dom.Document doc = db.parse(isr);
+			Element element = doc.getDocumentElement();
+
+			if(element != null && element.getNodeName() != null && element.getNodeName().contains(RESPONSE_XML_ERROR_NODE)) {
+				for(int n =0; n< element.getChildNodes().getLength(); n++) {
+					Node node = element.getChildNodes().item(n);
+			    	if(node.getNodeName() != null && node.getNodeName().contains(RESPONSE_XML_MESSAGE_NODE)) {
+			    		if(node.getChildNodes() != null) {
+			    			Node child = node.getFirstChild();			 
+			    			msg = child.getNodeValue();
+			    			exception =  new DocuWareException(msg, httpStatus);
+			    		}
+			    	}
+			    } 
+			}	
+		} catch (ParserConfigurationException e) {
+
+		} catch (SAXException e) {
+			
+		} catch (IOException e) {
+			
+		}
+		return exception;
+	}
+		
+	public static String getFilenameFromResponseHeader(Response response) {
+		String filename = null;
+		if(response != null) {
+			String disposition = response.getHeaderString(CONTENT_DISPOSITION);
+			filename = disposition.replaceFirst("(?i)^.*filename=\"?([^\"]+)\"?.*$", "$1");
+		}
+		return filename;
+	}
+		
+	public static File createPropertiesFile( List<DocuWareProperty> properties) throws IOException {
+		File propertiesFile = getUniquePropertiesFile();
+		DocuWareProperties docuWareproperties = new DocuWareProperties();
+		docuWareproperties.setProperties(properties);
+		FileUtils.write(propertiesFile.getJavaFile(), JsonUtils.serializeProperties(docuWareproperties), PROPERTIES_FILE_CHARSET);
+		return propertiesFile;
+	}
+	
+	private static Builder prepareRestClient(WebTarget target, DocuWareEndpointConfiguration configuration) {
+		return target.request().header("X-Requested-By", "ivy")
+				.header("MIME-Version", "1.0")
+				.header("Accept", "application/xml")
+				.header("Connection", "keep-alive")
+				.header("Authorization", AUTHORIZATION_BASIC + configuration.getCredentials());
```

Comment:
authorization should already be solved by the DocuwareAuthFeature. can you outline why this additional basic auth is required?

Reply:
yes, thats true. i have removed this from the header (and also from the configuration helper class)